File: Python/108. Convert Sorted Array to Binary Search Tree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    def __init__(self):
        self.nums = []

    def solve(self, a, b):
        if a == b:
            return None
        mid = (a + b) // 2
        root = TreeNode(self.nums[mid])
        root.left = self.solve(a, mid)
        root.right = self.solve(mid + 1, b)
        return root

    def sortedArrayToBST(self, nums: List[int]) -> Optional[TreeNode]:
        self.nums = nums
        return self.solve(0, len(nums))

File: C++/108. Convert Sorted Array to Binary Search Tree.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {

    vector<int> nums;

    TreeNode* solve(int a, int b)
    {
        if (a == b)
            return NULL;
        int mid = (a + b) / 2;
        TreeNode* root = new TreeNode(nums[mid]);
        root->left = solve(a, mid);
        root->right = solve(mid + 1, b);
        return root;
    }
public:
    TreeNode* sortedArrayToBST(vector<int>& vec) {
        nums = vec;
        return solve(0, nums.size());
    }
};